engine/src/test/java/io/zeebe/engine/processing/deployment/DeploymentReprocessingTest.java

```diff
@@ -33,82 +30,57 @@
 import org.junit.Rule;
 import org.junit.Test;
 
-public final class ReprocessingIssueDetectionMultiplePartitionTest {
+public final class DeploymentReprocessingTest {
 
-  private static final int PARTITION_COUNT = 2;
+  private static final int PARTITION_COUNT = 1;
+  private static final int DEPLOYMENT_DISTRIBUTION_ON_PROCESSING = 1;
 
   @Rule
   public final RecordingExporterTestWatcher recordingExporterTestWatcher =
       new RecordingExporterTestWatcher();
 
   private final DeploymentDistributorMock deploymentDistributorMock =
-      new DeploymentDistributorMock();
+      spy(new DeploymentDistributorMock());
 
   @Rule
   public final EngineRule engine =
       EngineRule.multiplePartition(PARTITION_COUNT)
           .withDeploymentDistributor(deploymentDistributorMock);
 
-  private long workflowInstanceKey;
-  private Record<JobRecordValue> jobCreated;
-
   @Before
   public void setup() {
-    // avoid that a deployment is distributed
-    deploymentDistributorMock.pushDeploymentCallback = CompletableActorFuture::new;
+    // complete the deployment distribution immediately to write the DISTRIBUTED event
+    deploymentDistributorMock.pushDeploymentCallback = () -> CompletableActorFuture.completed(null);
 
     engine
         .deployment()
-        .withXmlResource(
-            Bpmn.createExecutableProcess("process")
-                .startEvent()
-                .serviceTask("task", t -> t.zeebeJobType("test"))
-                .done())
+        .withXmlResource(Bpmn.createExecutableProcess("process").startEvent().done())
         .expectCreated()
         .deploy();
 
-    workflowInstanceKey = engine.workflowInstance().ofBpmnProcessId("process").create();
-    jobCreated = RecordingExporter.jobRecords(JobIntent.CREATED).getFirst();
-
-    assertThat(
-            RecordingExporter.records()
-                .limit(r -> r.getPosition() > jobCreated.getPosition())
-                .withValueType(ValueType.DEPLOYMENT))
-        .extracting(Record::getIntent)
-        .doesNotContain(DeploymentIntent.DISTRIBUTED);
+    RecordingExporter.deploymentRecords(DeploymentIntent.DISTRIBUTED).await();
 
     engine.stop();
+
+    // the deployment is distributed once on processing
+    verify(deploymentDistributorMock, times(DEPLOYMENT_DISTRIBUTION_ON_PROCESSING))
```

Comment:
We could reset the count after verification. What do you think?

Reply:
Good point. I applied your hint. Please have a look :eyes: